syntax = "proto3";

package nfb_grpc;

service Nfb {
	rpc Nfb_fdt_get(nfb_rpc_device) returns (nfb_fdt) {}
	rpc Nfb_comp_read(nfb_read_req) returns (nfb_read_resp) {}
	rpc Nfb_comp_write(nfb_write_req) returns (nfb_write_resp) {}
}

message nfb_rpc_device {
	string path = 1;
}

message nfb_fdt {
	bytes fdt = 1;
}

message nfb_read_req {
	int32 fdt_offset = 1;
	int32 nbyte = 2;
	int32 offset = 3;
}

message nfb_read_resp {
	int32 status = 1;
	bytes data = 2;
}

message nfb_write_req {
	int32 fdt_offset = 1;
	int32 nbyte = 2;
	int32 offset = 3;
	bytes data = 4;
}

message nfb_write_resp {
	int32 status = 1;
}
